core/src/collections.rs

```diff
@@ -0,0 +1,43 @@
+use crate::conn::MongoConnection;
+use crate::stmt::MongoStatement;
+use bson::{Array, Bson, Document};
+use mongodb::sync::Cursor;
+use std::error::Error;
+
+#[derive(Debug)]
+pub struct MongoCollections {
+    // The cursor on the result set.
+    resultset_cursor: Cursor<Document>,
+    // The current row the cursor points to.
+    current_row: Box<Array>,
```

Comment:
I planned to transform the Document in an Array (row) when calling next and then I would just get the cell at Array[col_index] from get_value(col_index). This mainly stems from the MongoStatement trait using an index for get_value. Maybe it would be better and easier to use names? I am not sure right now. I'll give it a bit more thought, but I have a feeling that it might make integration with the ODBC component easier to use names. The ODBC component will call get_value(name) for the field (=ODBC cell) it wants to retrieve from current Document(=ODBC row).
@pmeredit : WDYT?